### DISCUSSION TITLE
Paint user colours in system messages

### DISCUSSION THREAD

**Discussion by jupjohn** [Chatterino/chatterino2#5596] [Ideas] [answered]
Being able to display username colours in system messages (for things like bans, automod actions, etc.) would be nice from a UX perspective as users would be able to visually parse out username information from actions quicker. 

As part of this, I'd like to suggest a setting (MOAR SETTINGS) to allow for toggling between coloured and plain username rendering as some users may not want this. 

Based on Neri's comment in the c7 repository: https://github.com/SevenTV/chatterino7/issues/111#issuecomment-2350953289

**Comment by pajlada** [comment, marked as answer]
As for the suggestion as a whole: Yeah, it's just a matter of finding places where we don't color usernames & make them colored (and clickable). This will be done by replacing `TextElement` with usernames with the `MentionElement`.

The `Color @usernames` setting controls coloring of usernames both in system messages and in normal messages, I'm not sure I want to add a new setting that now controls colorizing of usernames in system messages.

**Reply by jupjohn** [reply]
That sounds like an even better solution; agreed that adding another setting is silly if it's already covered under an existing one 👍
